ui/src/components/tag/AddTagForm.js

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2021 Verizon Media
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import styled from '@emotion/styled';
+import InputLabel from '../denali/InputLabel';
+import Input from '../denali/Input';
+import { colors } from '../denali/styles';
+import Button from '../denali/Button';
+import Tag from '../denali/Tag';
+
+const SectionsDiv = styled.div`
+    width: 100%;
+    text-align: left;
+    background-color: ${colors.white};
+`;
+
+const SectionDiv = styled.div`
+    align-items: flex-start;
+    display: flex;
+    flex-flow: row nowrap;
+    padding: 10px 30px;
+`;
+
+const ContentDiv = styled.div`
+    flex: 1 1;
+    margin-right: 10px;
+`;
+
+const StyledInputLabel = styled(InputLabel)`
+    flex: 0 0 100px;
+    margin-right: 2%;
+`;
+
+const StyledInput = styled(Input)`
+    width: 500px;
+`;
+
+const ButtonDiv = styled.div`
+    margin-left: 10px;
+`;
+
+const StyledButton = styled(Button)`
+    width: 125px;
+`;
+
+const StyledIncludedValuesDiv = styled.div`
+    width: 65%;
+`;
+
+const StyledTag = styled(Tag)`
+    color: #d5d5d5;
+    &:hover {
+        background: #ffffff;
+    }
+    font-size: 14px;
+    height: 28px;
+    line-height: 14px;
+    margin: 5px 15px 5px 0;
+    padding: 6px 8px 7px 10px;
+`;
+
+const StyledTagColor = styled(StyledTag)`
+    background: rgba(53, 112, 244, 0.08);
+`;
+
+const StyledAnchor = styled.a`
+    text-decoration: none;
+`;
+
+const StyledAnchorActiveInline = { color: colors.linkActive };
+
+export default class AddTagForm extends React.Component {
```

Comment:
any reason we want to maintain state variables in this component?
a suggestion would be to just use this to render the form and use the props to populate values because currently you are doing setState here on input changed and again doing a setState in AddTag.js which is repetitive.

Reply:
AddTagForm is AddTag child.
On every change in AddTagForm (add tag, delete tag, change tag name) we need to inform AddTag since the submitting done through AddTag.
As a result both should keep state and not `a suggestion would be to just use this to render the form and use the props to populate values`

Reply:
Since addTagForm is child of AddTag, we can just update the state variable of AddTag by calling this.props.function()

This would update the state value in AddTag.js and this is what we pass to AddTagForm.js so AddTagForm gets the updates through props.

Reply:
other reason for keep it with state since addTagForm uses also for edit existing tag.
will check it again

Reply:
did you try removing state and moving to props?

Reply:
following up on this..tried moving to props from state ?